### PR TITLE
fix: replace antlr with regex to solve did creation issues

### PR DESCRIPTION
**Pull request by cristianIOHK**
# Overview
This PR replaces antlr with regex. This resolves [ATL-5818](https://input-output.atlassian.net/browse/ATL-5818).

Fixes #5818

### My PR contains...
* [ ] No code changes (changes to documentation, CI, metadata, etc.)
* [X] Bug fixes (non-breaking change which fixes an issue)
* [ ] Improvements (misc. changes to existing features)
* [ ] Features (non-breaking change which adds functionality)

### My changes...
* [ ] are breaking changes
* [X] are not breaking changes
* [ ] If yes to above: I have updated the documentation accordingly

### Documentation
* [X] My changes do not require a change to the project documentation
* [ ] My changes require a change to the project documentation
* [ ] If yes to above: I have updated the documentation accordingly

### Tests
* [ ] My changes can not or do not need to be tested
* [X] My changes can and should be tested by unit and/or integration tests
* [ ] If yes to above: I have added tests to cover my changes
* [ ] If yes to above: I have taken care to cover edge cases in my tests

[ATL-5818]: https://input-output.atlassian.net/browse/ATL-5818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by atala-dev** [comment]
### Code Coverage
|File|Coverage [63.58%]|
|:-|:-:|
|[atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/apollo/ApolloImpl.kt](https://github.com/input-output-hk/atala-prism-wallet-sdk-kmm/blob/9ca116ae62c92307d2152c77b1dae0511bb7ebc8/atala-prism-sdk%2Fsrc%2FcommonMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fwalletsdk%2Fapollo%2FApolloImpl.kt)|57.14%|
|[atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/apollo/utils/Secp256k1PrivateKey.kt](https://github.com/input-output-hk/atala-prism-wallet-sdk-kmm/blob/9ca116ae62c92307d2152c77b1dae0511bb7ebc8/atala-prism-sdk%2Fsrc%2FcommonMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fwalletsdk%2Fapollo%2Futils%2FSecp256k1PrivateKey.kt)|50.00%|
|[atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/castor/did/DIDParser.kt](https://github.com/input-output-hk/atala-prism-wallet-sdk-kmm/blob/9ca116ae62c92307d2152c77b1dae0511bb7ebc8/atala-prism-sdk%2Fsrc%2FcommonMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fwalletsdk%2Fcastor%2Fdid%2FDIDParser.kt)|81.82%|
|[atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/castor/did/DIDUrlParser.kt](https://github.com/input-output-hk/atala-prism-wallet-sdk-kmm/blob/9ca116ae62c92307d2152c77b1dae0511bb7ebc8/atala-prism-sdk%2Fsrc%2FcommonMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fwalletsdk%2Fcastor%2Fdid%2FDIDUrlParser.kt)|87.50%|
|[atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/outOfBand/OutOfBandInvitation.kt](https://github.com/input-output-hk/atala-prism-wallet-sdk-kmm/blob/9ca116ae62c92307d2152c77b1dae0511bb7ebc8/atala-prism-sdk%2Fsrc%2FcommonMain%2Fkotlin%2Fio%2Fiohk%2Fatala%2Fprism%2Fwalletsdk%2Fprismagent%2Fprotocols%2FoutOfBand%2FOutOfBandInvitation.kt)|75.00%|

|Total Project Coverage|26.20%|
|:-|:-:|

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/outOfBand/OutOfBandInvitation.kt [hunk 1]
```diff
@@ -32,8 +32,8 @@ constructor(
     @Serializable
     data class Body(
         @SerialName(GOAL_CODE)
-        val goalCode: String?,
-        val goal: String?,
+        val goalCode: String? = "",
+        val goal: String? = "",
```
Comment (hamada147) [inline atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/outOfBand/OutOfBandInvitation.kt:36]: Should these values be null and with `@EncodeDefault` or as per the conversation on slack it should be empty string?!

**Review by hamada147 — APPROVED** [review]
LGTM but have one question shared on the change.

File: atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/outOfBand/OutOfBandInvitation.kt (hunk 1 above)
Comment (hamada147) [inline atala-prism-sdk/src/commonMain/kotlin/io/iohk/atala/prism/walletsdk/prismagent/protocols/outOfBand/OutOfBandInvitation.kt:36]:
```suggestion
        @EncodeDefault
        val goalCode: String? = null,
        @EncodeDefault
        val goal: String? = null,
```

**Review by hamada147 — APPROVED** [review]
